File: 485/a.java
public class a {

    public static void main(String[] args) {
        java.util.Scanner sc = new java.util.Scanner(System.in);
        int a = sc.nextInt(), m = sc.nextInt();
        sc.close();

        for (int i = 0; i <= m; i++) {
            a = (a * 2) % m;
        }

        System.out.println(a == 0 ? "Yes" : "No");
    }
}

File: 485/a.cpp
#include <cstdio>

int main() {
  int a, m;
  scanf("%d%d", &a, &m);

  for (int i = 0; i <= m; ++i) {
    a *= 2, a %= m;
  }

  printf(a == 0 ? "Yes\n" : "No\n");
  return 0;
}
